controller/update.go

```diff
@@ -321,9 +321,9 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
 				// we increase the addInProgress counter before starting the goroutine
 				// to avoid a race condition in the allowed calculation
 				atomic.AddInt64(&addInProgress, 1)
-				go func(ctx context.Context) {
-					ctx = context.WithValue(ctx, "add slice", sliceID)
-					c.Config.Logger.Debug(ctx, "controller: starting to add slice: %v", sliceID)
+				go func(req Request) {
+					ctx := context.WithValue(ctx, "add slice", req.SliceIDs)
+					c.Config.Logger.Debug(ctx, "controller: starting to add slice: %v", req.SliceIDs)
 
 					newSliceIDs, err := c.addFirst(ctx, newReq, opts)
```

Comment:
The whole point of this fix is actually passing the request properly to the goroutine AND using it. So `newReq` -> `req`. The issues was not the logging, but the potential raise conditions during updates.